File: back/src/main/java/com/back/global/security/oauth2/info/GoogleOAuth2UserInfo.java
package com.back.global.security.oauth2.info;

import java.util.Map;

public class GoogleOAuth2UserInfo implements OAuth2UserInfo {
    private final Map<String, Object> attributes;

    public GoogleOAuth2UserInfo(Map<String, Object> attributes) {
        this.attributes = attributes;
    }

    @Override
    public String getProviderId() {
        Object sub = attributes.get("sub");
        return sub == null ? null : sub.toString();
    }

    @Override
    public String getEmail() {
        Object email = attributes.get("email");
        return email == null ? null : email.toString();
    }

    @Override
    public String getName() {
        Object name = attributes.get("name");
        return name == null ? "구글사용자" : name.toString();
    }
}

File: back/src/main/kotlin/com/back/global/security/oauth2/info/GoogleOAuth2UserInfo.kt
package com.back.global.security.oauth2.info

class GoogleOAuth2UserInfo(
    private val attributes: Map<String, Any>
) : OAuth2UserInfo {

    override val providerId: String?
        get() = attributes["sub"]?.toString()

    override val email: String?
        get() = attributes["email"]?.toString()

    override val name: String
        get() = attributes["name"]?.toString() ?: "구글사용자"

    override val isEmailVerified: Boolean
        get() = attributes["email_verified"]?.toString()?.toBooleanStrictOrNull() == true
}
